### PR TITLE
Add NonTransactional Support

### PR DESCRIPTION
**Pull request by dtelaroli**
PR related to https://github.com/garcia-jj/vraptor-plugin-hibernate4/issues/12

### CODE REVIEW

File: src/main/java/br/com/caelum/vraptor/plugin/hibernate4/HibernateTransactionInterceptor.java [hunk 1]
```diff
@@ -58,6 +58,10 @@ public void intercept(InterceptorStack stack, ResourceMethod method, Object inst
     }
 
     public boolean accepts(ResourceMethod method) {
-        return true; // Will intercept all requests
+        return noHasAnnotationNonTransactional(method);
     }
+
```
Comment (garcia-jj) [inline src/main/java/br/com/caelum/vraptor/plugin/hibernate4/HibernateTransactionInterceptor.java:63]: rename to `hasNonTransactionAnnotation`, and change the code to return if method has the annotation. So in the `accepts` method use a negate condition. I think that with this change, the code will more easily to understood.

File: src/main/java/br/com/caelum/vraptor/plugin/hibernate4/NonTransactional.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package br.com.caelum.vraptor.plugin.hibernate4;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
```
Comment (garcia-jj) [inline src/main/java/br/com/caelum/vraptor/plugin/hibernate4/NonTransactional.java:12]: Please, write some docs about this annotation.

File: src/main/java/br/com/caelum/vraptor/plugin/hibernate4/NonTransactional.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+@Target(ElementType.METHOD)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface NonTransactional {
+	boolean managed() default false;
```
Comment (garcia-jj) [inline src/main/java/br/com/caelum/vraptor/plugin/hibernate4/NonTransactional.java]: where `managed` attribute is used?

Comment (dtelaroli) [inline src/main/java/br/com/caelum/vraptor/plugin/hibernate4/NonTransactional.java]: ctrl c ctrl v
sorry

File: src/main/java/br/com/caelum/vraptor/plugin/hibernate4/HibernateTransactionInterceptor.java (hunk 1 above)
Comment (garcia-jj) [inline src/main/java/br/com/caelum/vraptor/plugin/hibernate4/HibernateTransactionInterceptor.java:63]: or you can rename method to `isNonTransactional`. may be more clear.

Comment (dtelaroli) [inline src/main/java/br/com/caelum/vraptor/plugin/hibernate4/HibernateTransactionInterceptor.java:63]: I agree

File: src/main/java/br/com/caelum/vraptor/plugin/hibernate4/HibernateTransactionInterceptor.java [hunk 4]
```diff
@@ -58,6 +58,10 @@ public void intercept(InterceptorStack stack, ResourceMethod method, Object inst
     }
 
     public boolean accepts(ResourceMethod method) {
-        return true; // Will intercept all requests
+        return !isNonTransactional(method);
     }
+
+	private boolean isNonTransactional(ResourceMethod method) {
+		return method.getMethod().getAnnotation(NonTransactional.class) != null;
```
Comment (lucascs) [inline src/main/java/br/com/caelum/vraptor/plugin/hibernate4/HibernateTransactionInterceptor.java]: you can use: `method.containsAnnotation(NonTransactional.class)`

**Comment by garcia-jj** [comment]
Merged. Thank you.

**Comment by dtelaroli** [comment]
@garcia-jj 

Can you provide this feature in the hibernate 4.2.x version?
The plugin 1.1.5 is not working with jboss 7.1.1.

**Comment by garcia-jj** [comment]
Yes. I will check this tonight.

**Comment by garcia-jj** [comment]
Changes applyed here: https://github.com/garcia-jj/vraptor-plugin-hibernate4/tree/hibernate-4.2

**Comment by dtelaroli** [comment]
Perfect. Can you deploy a version on maven?

I thought that you would downgrade and do a cherry pick to do that. But you can deploy it with a classifier.

**Comment by lucascs** [comment]
No need for a different version. It's not enabled by default, you have to register the decorator on beans.xml

**Comment by Turini** [comment]
@lucascs, I think you are talking about `vraptor-jpa` transactional feature.
This is the vraptor 3 version of `vraptor-plugin-hibernate4` =P 
(there is no decorator and beans.xml file... I think we really need to deploy)

**Comment by lucascs** [comment]
ops =S

**Pull request merged**